packages/vector_graphics_codec/lib/vector_graphics_codec.dart

```diff
@@ -0,0 +1,638 @@
+    buffer._putFloat64(y1);
+    buffer._putFloat64(x2);
+    buffer._putFloat64(y2);
+    buffer._putFloat64(x3);
+    buffer._putFloat64(y3);
+  }
+
+  /// Write a close command to the current path.
+  ///
+  /// Throws a [StateError] if there is not an active path.
+  void writeClose(VectorGraphicsBuffer buffer) {
+    if (buffer._currentPathId == -1) {
+      throw StateError('There is no active Path.');
+    }
+    buffer._putUint8(_closeTag);
+  }
+
+  /// Finishes building the current path
+  ///
+  /// Throws a [StateError] if there is not an active path.
+  void writeFinishPath(VectorGraphicsBuffer buffer) {
+    if (buffer._currentPathId == -1) {
+      throw StateError('There is no active Path.');
+    }
+    buffer._currentPathId = -1;
+  }
+
+  void _readPath(_ReadBuffer buffer, VectorGraphicsCodecListener? listener) {
+    final int fillType = buffer.getUint8();
+    final int id = buffer.getInt32();
+    listener?.onPathStart(id, fillType);
+  }
+
+  void _readMoveTo(_ReadBuffer buffer, VectorGraphicsCodecListener? listener) {
+    final double x = buffer.getFloat64();
+    final double y = buffer.getFloat64();
+    listener?.onPathMoveTo(x, y);
+  }
+
+  void _readLineTo(_ReadBuffer buffer, VectorGraphicsCodecListener? listener) {
+    final double x = buffer.getFloat64();
+    final double y = buffer.getFloat64();
+    listener?.onPathLineTo(x, y);
+  }
+
+  void _readCubicTo(_ReadBuffer buffer, VectorGraphicsCodecListener? listener) {
+    final double x1 = buffer.getFloat64();
+    final double y1 = buffer.getFloat64();
+    final double x2 = buffer.getFloat64();
+    final double y2 = buffer.getFloat64();
+    final double x3 = buffer.getFloat64();
+    final double y3 = buffer.getFloat64();
+    listener?.onPathCubicTo(x1, y1, x2, y2, x3, y3);
+  }
+
+  void _readClose(_ReadBuffer buffer, VectorGraphicsCodecListener? listener) {
+    listener?.onPathClose();
+  }
+
+  void _readDrawPath(
+      _ReadBuffer buffer, VectorGraphicsCodecListener? listener) {
+    final int pathId = buffer.getInt32();
+    final int paintId = buffer.getInt32();
+    listener?.onDrawPath(pathId, paintId);
+  }
+
+  void _readDrawVertices(
+      _ReadBuffer buffer, VectorGraphicsCodecListener? listener) {
+    final int paintId = buffer.getInt32();
+    final int verticesLength = buffer.getInt32();
+    final Float32List vertices = buffer.getFloat32List(verticesLength);
+    final int indexLength = buffer.getInt32();
+    Uint16List? indices;
+    if (indexLength != 0) {
+      indices = buffer.getUint16List(indexLength);
+    }
+    listener?.onDrawVertices(vertices, indices, paintId);
+  }
+}
+
+/// Implement this listener class to support decoding of vector_graphics binary
+/// assets.
+abstract class VectorGraphicsCodecListener {
+  /// A paint object has been decoded.
+  ///
+  /// If the paint object is for a fill, then [strokeCap], [strokeJoin],
+  /// [strokeMiterLimit], and [strokeWidget] will be `null`.
+  void onPaintObject({
```

Comment:
nit: maybe just `onStroke` and `onFill`?

Reply:
done